code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -469,6 +476,64 @@
 		for(var/obj/structure/window/W in T.contents)
 			W.take_damage(rand(80, 100))
 
+/obj/effect/proc_holder/spell/targeted/click/mindshield_melt //melts mindshields
+	name = "Melt Mindshield"
+	desc = "Allows you to destroy a mindshield."
+	panel = "Shadowling Abilities"
+	charge_max = 600
+	clothes_req = FALSE
+	range = 1 //Adjacent to user
+	action_icon_state = "mindshield_melt"
+
+	click_radius = -1 // Precision baby
+	selection_activated_message		= "<span class='notice'>Your collect your mind to destroy a mindshield. <B>Left-click to cast at a target!</B></span>"
+	selection_deactivated_message	= "<span class='notice'>Your mind relaxes.</span>"
+	allowed_type = /mob/living/carbon/human
+
+/obj/effect/proc_holder/spell/targeted/click/mindshield_melt/can_cast(mob/user = usr, charge_check = TRUE, show_message = FALSE)
+	if(!shadowling_check(user))
+		return FALSE
+	return ..()
+
+/obj/effect/proc_holder/spell/targeted/click/mindshield_melt/valid_target(mob/living/carbon/human/target, user)
+	if(!..())
+		return FALSE
+	return target.key && target.mind && !target.stat && !is_shadow_or_thrall(target) && target.client
+
+/obj/effect/proc_holder/spell/targeted/click/mindshield_melt/cast(list/targets, mob/user = usr)
+	var/mob/living/carbon/human/ling = user
+	if(!(ling.mind in SSticker.mode.shadows))
+		return
+	var/mob/living/carbon/human/target = targets[1]
+	if(ismindshielded(target))
```

Comment:
I'd probably add this check to `valid_target`, so that the spell isn't cast and wasted on a non-mindshielded person.